client/signup/main.jsx

```diff
@@ -149,7 +151,14 @@ const Signup = React.createClass( {
 			defer( function() {
 				page( destination );
 			}.bind( this ) );
-		} else {
+		}
+
+		if ( ! user.get() && config.isEnabled( 'oauth' ) ) {
```

Comment:
We could keep this condition inside the `else` and just create a new `if/else` branch to check `config.isEnabled( 'oauth' )`. Or break out into separate functions. Probably easier to follow than the multiple if checks here?